src/Schema/DataObject/InheritanceBuilder.php

```diff
@@ -66,12 +71,14 @@ public function fillAncestry(ModelType $modelType): void
                 ? $fieldObj->getPropertyName()
                 : $fieldObj->getName();
             // If the field is unique to the descendant, skip it.
-            if ($parentModel->getModel()->hasField($fieldName)) {
+            if ($ancestorModel->getModel()->hasField($fieldName)) {
                 $clone = clone $fieldObj;
-                $parentModel->addField($fieldObj->getName(), $clone);
+                $ancestorModel->addField($fieldObj->getName(), $clone);
             }
         }
-        $this->fillAncestry($parentModel);
+        if ($modelType->getName() !== $ancestorModel->getName()) {
+            $this->fillAncestry($ancestorModel);
+        }
```

Comment:
```suggestion
        $this->fillAncestry($ancestorModel);
```
The changes above mean we won't ever get in this scenario by the time we get down here, which makes the flow a little cleaner.

Reply:
Not changing logic above, so not changing here either

Reply:
done